src/hotspot/share/opto/cfgnode.cpp

```diff
@@ -1472,13 +1472,17 @@ Node* PhiNode::Identity(PhaseGVN* phase) {
   // It would check for a tributary phi on the backedge that the main phi
   // trivially, perhaps with a single cast.  The unique_input method
   // does all this and more, by reducing such tributaries to 'this'.)
-  Node* uin = unique_input(phase, false);
-  if (uin != nullptr) {
-    return uin;
+  {
+    Node* uin = unique_input(phase, false);
+    if (uin != nullptr) {
+      return uin;
+    }
   }
-  uin = unique_input_recursive(phase);
-  if (uin != nullptr) {
-    return uin;
+  {
+    Node* uin = unique_constant_input_recursive(phase);
```

Comment:
I think the original format was fine but I have no strong opinion.

Reply:
Unnecessary to make it compile. But I still think it's helpful: each block can be rearranged since it doesn't change the semantics of an outer-living variable. It makes explicit that `uin` isn't used after... But I said I won't fight too hard, so I reverted!